Update the order notes design

0nko opened the pull request:
Fixes #1452. This PR updates the of the order notes in the order detail screen to match the new design.

<img src="https://user-images.githubusercontent.com/1522856/66655570-94ca7580-ec3c-11e9-8754-097f22870356.png" width="300" />

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

AmandaRiu commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/notes/OrderDetailOrderNoteItemView.kt:
```diff
@@ -1,39 +1,46 @@
-import org.wordpress.android.fluxc.model.WCOrderNoteModel
 
-class OrderDetailOrderNoteItemView @JvmOverloads constructor(ctx: Context, attrs: AttributeSet? = null)
+class OrderDetailOrderNoteItemView @JvmOverloads constructor(val ctx: Context, attrs: AttributeSet? = null)
```
Why the added `val` here? It's just creating another variable for the `context` variable which is available from the parent ConstraintLayout. If you remove the `val` , you can just use `context` in place of `ctx` in the class body. Example:

Instead of `note.isCustomerNote -> ctx.getString(R.string.orderdetail_note_public)`
Use: `note.isCustomerNote -> context.getString(R.string.orderdetail_note_public)`

kyleaparker commented:
>Also I'd like to double-check the changes to line spacing. Using 1.5 as the line multiplier has made a really big difference in how much space a note takes, for example:

We are going to be moving towards using system defaults for typography. For body 2 (font size 14px) the line height should be 20px.

AmandaRiu commented:
@0nko this PR now has some conflicts that need to be resolved

0nko commented:
Thanks for the review @AmandaRiu and sorry for the delayed response.. I've been trying to fiddle with the connected tests but without luck. Actually, now there are new (unrelated) failing tests that came from `develop`. What do you think about fixing all of them in a separate PR?

> Looking at the designs, I don't see anything like (System) or (Private) in the order notes section, was this already cleared by design? 

They're in Figma. See the #1452 screenshot. 

> Using 1.5 as the line multiplier has made a really big difference in how much space a note takes

I've changed it to 1.4. I'm not sure if the line height is exactly 20px as per @kyleaparker's design, but it looks pretty close.

I think I've addressed all the other comments.

AmandaRiu commented:
@0nko finished with another round:

> What do you think about fixing all of them in a separate PR?

works for me 👍 

> They're in Figma. See the #1452 screenshot.

Am I missing something? I don't see (System) or (Private) in that screenshot. Here's the relevant snippet:

![Screen Shot 2019-10-18 at 3 53 07 PM](https://user-images.githubusercontent.com/5810477/67132932-6e1db900-f1bf-11e9-859c-8ca5c5d94b37.png)

Personally, I think the (System) and (Private) labels make perfect sense. I just want to make sure. 

One last thing - after playing around I've figured out you'd want a line multiplier of `1.2` in order to get close to a line height of 20px. I only know this because `android:lineHeight="20dp"` is supported in newer API versions (but not all the ones we support). Here is the result:

![Screen Shot 2019-10-18 at 3 55 58 PM](https://user-images.githubusercontent.com/5810477/67133080-16338200-f1c0-11e9-8106-8180fb51634d.png)

Thanks!
